ideascube/utils.py

```diff
@@ -87,3 +88,7 @@ def get_file_sha256(path):
 
 def get_file_size(path):
     return os.stat(path).st_size
+
+def tag_splitter(tag_string):
+    tags = set(t.strip().lower() for t in re.split('[;,]+', tag_string) if t.strip())
```

Comment:
> We could not have a Tag(name="Bar", slug="bar") and a Tag(name="bar", slug="bar") at the same time.

I know, and that's a good thing.

The reason I had suggested lower-casing the tags in the first place was to deduplicate them.That is, if someone entered "Cinema" as a tag for a media document, and then "cinema" as a tag for another document, the goal was to consider them the same tag.

You're right that the user experience this provides is tricky (if not impossible) to get right. It has the exact same problems as I was complaining with your lower-casing everything.

Therefore, let's not lower-case at all.